vault_service/views/user.py

```diff
@@ -195,4 +197,353 @@ def store_data():
+                      'classes': setup.classes,
+                      'data': setup.data,
+                      'created': setup.created.isoformat(),
+                      'updated': setup.updated.isoformat()}
+
+        return json.dumps(output), 200
+
+
+@advertise(scopes=['ads-consumer:myads'], rate_limit = [1000, 3600*24])
+@bp.route('/get-myads/<user_id>', methods=['GET'])
+def get_myads(user_id):
+    '''
+    Fetches a myADS profile for the pipeline for a given uid
+    '''
+
+    # takes a given uid, grabs the user_data field, checks for the myADS key - if present, grabs the appropriate
+    # queries/data from the queries/myADS table. Finally, returns the appropriate info along w/ the rest
+    # of the setup from the dict (e.g. active, frequency, etc.)
+
+    # TODO check rate limit
+
+    # structure in vault:
+    # "myADS": [{"name": user-supplied name, "qid": QID from query table (type="query") or ID from myads table
+    # (type="template"), "type":  "query" or "template", "active": true/false, "frequency": "daily" or "weekly",
+    # "stateful": true/false}]
+
+    output = []
+    with current_app.session_scope() as session:
+        setups = session.query(MyADS).filter_by(user_id=user_id).filter_by(active=True).order_by(MyADS.id.asc()).all()
+        if not setups:
+            return '{}', 404
+        for s in setups:
+            o = {'id': s.id,
+                 'name': s.name,
+                 'type': s.type,
+                 'active': s.active,
+                 'stateful': s.stateful,
+                 'frequency': s.frequency,
+                 'template': s.template,
+                 'classes': s.classes,
+                 'data': s.data,
+                 'created': s.created.isoformat(),
+                 'updated': s.updated.isoformat()}
+
+            if s.type == 'query':
+                q = session.query(Query).filter_by(id=s.query_id).first()
+                qid = q.qid
+            else:
+                qid = None
+            o['qid'] = qid
+
+            output.append(o)
+
+    return json.dumps(output), 200
+
+
+@advertise(scopes=['ads-consumer:myads'], rate_limit = [1000, 3600*24])
+@bp.route('/myads-users/<iso_datestring>', methods=['GET'])
+def export(iso_datestring):
+    '''
+    Get the latest changes (as recorded in users table)
+        The optional argument latest_point is RFC3339, ie. '2008-09-03T20:56:35.450686Z'
+    '''
+    # TODO check rate limit
+    # TODO need to add a created/update field to users table
+
+    # inspired by orcid-service endpoint of same endpoint, checks the users table for profiles w/ a myADS setup that
+    # have been updated since a given date/time; returns these profiles to be added to the myADS processing
+
+    latest = parser.parse(iso_datestring) # ISO 8601
+    output = []
+    with current_app.session_scope() as session:
+        setups = session.query(MyADS).filter(MyADS.updated > latest).order_by(MyADS.updated.asc()).all()
```

Comment:
Deleted setups? I'm not sure I understand

Reply:
If I delete a notification/alert/setup (not sure what's the right nomenclature), from the code I see that we delete the entry (we do not mark it as deleted but we really remove it). So, any "show me what's new since that date" will not return anything saying "this setup was deleted and you should not generate notifications anymore for this". Or am I missing something?

Reply:
Ah, I understand. We're not actually fetching any setups here, just any user_ids that might have changed. In the pipeline, there's a table that stores the user_ids of everyone we've run myADS for before, and that list is then combined with the output from this endpoint. At that point, we check each user_id in that combined list for their current myADS setups via the get_myads endpoint (so the deleted setups are excluded then)

Reply:
Great! Then i didn't say anything ;-)